### ISSUE TITLE
connector secret is getting deleted

### ISSUE THREAD

**Issue by KouriR** [wiz-sec/charts#62]
I am using ArgoCD (v2.5.2) to deploy the wiz-kubernetes-connector:1.0.0 chart.

I have the following values set:
```
wizApiToken:
  # Specifies whether an api token secret should be created
  # If create is false you need to create it with clientId, clientToken
  createSecret: false
```

Upon syncing, the helm pre-install/pre-upgrade hook fires, which creates a job which creates a `wiz-kubernetes-connector.connectorSecretName` secret that the wiz-kubernetes-connector-broker pod mounts when it starts up. However, when the job finishes, the secret is being deleted before the pod is finished creating, so the pod cannot start up.

Expected behavior: The `wiz-kubernetes-connector.connectorSecretName` secret shouldn't be deleted, as this prevents the pod from starting up.

**Comment by KouriR** [comment]
I have found that the connector secret being deleted is due to having "Prune Resources" enabled in my ArgoCD app sync policy.

The secret is initially created during the pre-install/pre-upgrade hook, but when the connector-creator job runs, the secret seems to be recreated without the labels added by Helm, so ArgoCD doesn't see the resource as being associated with the chart and wants to prune it.

I think this can be fixed with two changes:
1) Have the connector-creator job create the secret if it doesn't exist, so the job doesn't fail if the resource is not there initially.
2) Create the secret with the necessary helm labels so that it doesn't get auto-pruned, e.g.:
```
  labels:
    app.kubernetes.io/instance: wiz-kubernetes-connector
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: wiz-kubernetes-connector
    app.kubernetes.io/version: 1.0.135029
    helm.sh/chart: wiz-kubernetes-connector-1.0.0
```

**Comment by lir-wiz** [comment]
Thanks for bringing this to our attention @KouriR !
Our engineers are working on a fix. We will keep you updated.

**Comment by lir-wiz** [comment]
@KouriR this should be fixed now.

**Comment by KouriR** [comment]
Hi @lir-wiz, this seems to be fixed in chart 1.0.5, thanks!
